### PR TITLE
World Select screen, Level Select screens, and level integration

### PR DESCRIPTION
**Pull request by BryanGraeser**
Implemented:
- World select screen
- museum, city, and space level select screens (just click on their buttons from world select) -- **note that other worlds (island, paris, etc) are NOT connected bc we don't have levels yet, so clicking on them will break the game** <-- I will disable these buttons before we submit
image
image

Museum = easy
City = medium
Space = hard

Please test and make sure you can play all the levels and test on AT LEAST android before presentation demo.

**Known Issues:**
1. For some reason clicking on a button causes all buttons before it to appear dropped pressed down; idk why. Something funky is going on.
image

2. Safe space/resizability. Currently locking reasonable resolution with following code to have some semblance of resizability for mobile, but it is causing safe space issues (including making back button and menu button go partially or fully off screen)
``` #define SCENE_SIZE 1024
// Lock the scene to a reasonable resolution
    if (screenSize.width > screenSize.height) {
        screenSize *= SCENE_SIZE / screenSize.width;
    } else {
        screenSize *= SCENE_SIZE / screenSize.height;
    }```

3. Decent amount of hard-coding, likely some reusable assets that can be factored out (like indiv level select json's but it is fine for now)

### CODE REVIEW

File: source/PPApp.cpp [hunk 1]
```diff
@@ -11,6 +11,7 @@ void PanicPainterApp::onStartup() {
     _assets->attach<Texture>(TextureLoader::alloc()->getHook());
     _assets->attach<Sound>(SoundLoader::alloc()->getHook());
     _assets->attach<SceneNode>(Scene2Loader::alloc()->getHook());
+    _assets->attach<WidgetValue>(WidgetLoader::alloc()->getHook());
```
Comment (BryanGraeser) [inline source/PPApp.cpp:14]: @foxhatleo this took me so much time to figure out why widgets weren't working... Is there a reason we removed this line when we copied App.cpp initially?

File: source/scenes/level/PPLevelSelectScene.cpp [hunk 2]
```diff
@@ -4,37 +4,24 @@
...
     // Lock the scene to a reasonable resolution
-    if (screenSize.width > screenSize.height) {
-        screenSize *= SCENE_SIZE / screenSize.width;
-    } else {
-        screenSize *= SCENE_SIZE / screenSize.height;
+    if (_screenSize.width > _screenSize.height) {
+        _screenSize *= SCENE_SIZE / _screenSize.width;
+    }
+    else {
+        _screenSize *= SCENE_SIZE / _screenSize.height;
```
Comment (BryanGraeser) [inline source/scenes/level/PPLevelSelectScene.cpp:14]: Hacky fix, need better resizability

File: source/scenes/level/PPWorldSelectScene.cpp [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    // Initialize the scene to a locked width
+    Size screenSize = Application::get()->getDisplaySize();
+
+    // Lock the scene to a reasonable resolution
+    if (screenSize.width > screenSize.height) {
+        screenSize *= SCENE_SIZE / screenSize.width;
+    } else {
+        screenSize *= SCENE_SIZE / screenSize.height;
+    }
+
```
Comment (BryanGraeser) [inline source/scenes/level/PPWorldSelectScene.cpp:15]: Hacky fix, need better resizability

**Review by lsizemore8 — CHANGES_REQUESTED** [review]
Idk if this will get fixed in time/it may just be worth merging this in for presentation and dealing with it after presentation but when you choose a specific world (ie go to level select) the back button is not on the screen at all.

**Review by lsizemore8 — APPROVED** [review]
It's got problems but it works for today

**Pull request merged**